sophos_central/plugin.spec.yaml

```diff
@@ -0,0 +1,527 @@
+      type: string
+    result:
+      type: string
+      enum:
+      - NOT_APPLICABLE
+      - SUCCESS
+      - NOT_FOUND
+      - DELETED
+      - FAILED_TO_DELETE
+      - WHITELISTED
+      - OTHER_ERROR
+      - FAILED_TO_DELETE_SYSTEM_PROTECTED
+    type:
+      type: string
+  installer_info:
+    beta:
+      type: boolean
+    command:
+      type: string
+    platform:
+      type: string
+    productName:
+      type: string
+    url:
+      type: string
+  customer_featuresResponse:
+    features:
+      type: '[]customer_feature'
+  core_remedy_items:
+    items:
+      type: '[]core_remedy_item'
+    totalItems:
+      type: integer
+  legacy_event_entity:
+    core_remedy_items:
+      description: details of the items cleaned or restored
+      type: core_remedy_items
+    created_at:
+      type: string
+      description: The date at which the event was created
+    customer_id:
+      type: string
+      description: The identifier of the customer for which record is created
+    endpoint_id:
+      type: string
+      description: The corresponding endpoint id associated with the record
+    endpoint_type:
+      type: string
+      description: The corresponding endpoint type associated with the record
+    group:
+      type: string
+      description: The group associated with the group
+    id:
+      type: string
+      description: The Identifier for the event
+    location:
+      type: string
+      description: The location captured for this record
+    name:
+      type: string
+      description: The name of the record created
+    origin:
+      type: string
+      description: originating component of a detection
+    severity:
+      type: string
+      description: The severity for this alert
+      enum:
+      - NONE
+      - LOW
+      - MEDIUM
+      - HIGH
+      - CRITICAL
+    source:
+      type: string
+      description: The source for this record
+    threat:
+      type: string
+      description: The threat associated with the record
+    type:
+      type: string
+      description: The type of this record
+    user_id:
+      type: string
+      description: The identifier of the user for which record is created
+    when:
+      type: string
+      description: The date at which the event was created
+    whitelist_properties:
+      type: '[]endpoint_whitelist_properties'
+      description: The properties by which this event can be whitelisted on an endpoint,
+        if applicable
+  event_aggregate:
+    has_more:
+      type: boolean
+    items:
+      type: '[]legacy_event_entity'
+    next_cursor:
+      type: string
+      description: Value of the next cursor. This will be used to make next call of API
+  hashes_response:
+    hashes:
+      type: object
+  installer_info_response:
+    installer_info:
+      type: '[]installer_info'
+
+connection:
+  url:
+    title: URL
+    description: Host URL
+    type: string
+    required: true
+    example: https://api-eu02.central.sophos.com
```

Comment:
@r7-mgorny Can we add an enum here with a list of the supported URLs

```

Data Region | API Host
-- | --
US (West) | https://api-us01.central.sophos.com
US (East) | https://api-us03.central.sophos.com
EU (Ireland) | https://api-eu01.central.sophos.com
DE (Germany) | https://api-eu02.central.sophos.com
```

https://developer.sophos.com/intro

Reply:
Actually, it looks like you might be able to auto-discover the region without needing the `url` input.

> As a tenant, you can call GET /whoami/v1 to find your own data region.

https://developer.sophos.com/intro

Reply:
I was tried to get this from whoami, but this is response for partner API:
```
{'id': 'ff5f40e9-220a-41e7-b530-e200bb8b2ff6', 'idType': 'partner', 'apiHosts': {'global': 'https://api.central.sophos.com', 'dataRegion': None}}
```

I will add enum.